### PR TITLE
Support GitHub Codespaces

### PR DESCRIPTION
**Pull request by victoriadrake**
This PR covers issue #546.

- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Add configuration settings and .devcontainer from https://github.com/microsoft/vscode-dev-containers
- Add contribution guidance note

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "editor.formatOnSave": true,
```
Comment (kingthorin) [inline .vscode/settings.json:2]: This seems like something that will introduce unrelated changes and make thing harder to review.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    "editor.formatOnType": true,
+    "files.autoSave": "afterDelay",
+    "files.autoSaveDelay": 1500,
+    "[markdown]": {
+        "editor.wordWrap": "bounded",
+        "editor.wordWrapColumn": 100,
+        "editor.quickSuggestions": false,
+    },
```
Comment (kingthorin) [inline .vscode/settings.json:10]: Unless we can introduce this formatting requirement for the GH WebUI, VSCode, Eclipse, etc then setting it for one is a mistake.

Considering we have no formatting established thus far I'd suggest removing for now.

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    "markdown.extension.toc.levels": "2..6",
+    "markdown.preview.fontSize": 21,
+    "markdownlint.config": {
+        "code-block-style": { "style" : "fenced" },
+        "line-length": false,
+        "no-duplicate-heading": false,
+        "no-trailing-punctuation": { "punctuation" : ".,;:!。，；：!" },
+    },
```
Comment (kingthorin) [inline .vscode/settings.json:18]: Can't we somehow leverage the existing file instead of trying to keep this in-sync across multiple resources?

https://github.com/OWASP/wstg/blob/master/.markdownlint.json

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -4,14 +4,14 @@ Thank you for considering contributing to the Web Security Testing Guide (WSTG)!
...
-  - [How to Open an Issue](#how-to-open-an-issue)
-  - [How to Submit a Pull Request](#how-to-submit-a-pull-request)
-  - [How to Set Up Your Contributor Environment](#how-to-set-up-your-contributor-environment)
+- [Become an Author](#become-an-author)
+- [Become a Reviewer or Editor](#become-a-reviewer-or-editor)
+  - [Technical Review](#technical-review)
+  - [Editorial Review](#editorial-review)
+- [How to Open an Issue](#how-to-open-an-issue)
+- [How to Submit a Pull Request](#how-to-submit-a-pull-request)
+- [Contributing with Codespaces](#contributing-with-codespaces)
```
Comment (kingthorin) [inline CONTRIBUTING.md:13]: I'd suggest that this should be the last section, until we see how popular/convenient it becomes.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -78,6 +78,14 @@ To increase the chances that your PR is merged, please make sure that:
 
 Once the PR is complete, we'll merge it! At that point, you may like to add yourself to [the project's list of authors, reviewers, or editors](document/1-Frontispiece/README.md).
 
+## Contributing with Codespaces
```
Comment (kingthorin) [inline CONTRIBUTING.md:81]: I'd suggest that this should be the last section, until we see how popular/convenient it becomes.

File: .vscode/settings.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (kingthorin) [inline .vscode/settings.json:1]: It seems like we can avoid adding IDE specific files to the repo by including the stuff in this file as part of devcontainer.json.

https://docs.github.com/en/github/developing-online-with-codespaces/configuring-codespaces-for-your-project#about-default-codespace-configurations

File: .vscode/settings.json (hunk 2 above)
Comment (victoriadrake) [reply .vscode/settings.json:10]: This isn't formatting, but settings for when you're using the editor. Do you still think we should omit them?

Comment (kingthorin) [reply .vscode/settings.json:10]: If it's just softwrap and isn't actually making format changes then I'm good with it.

File: .vscode/settings.json (hunk 3 above)
Comment (victoriadrake) [reply .vscode/settings.json:18]: Possibly, but I'll have to investigate further: https://code.visualstudio.com/docs/remote/containers-advanced

I'll remove these for now.

File: .devcontainer/devcontainer.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
+{
+    "name": "Markdown Editing",
+    "dockerFile": "Dockerfile",
+    // Set *default* container specific settings.json values on container create.
+    "settings": {
+        "terminal.integrated.shell.linux": "/bin/bash"
+
+            "editor.formatOnType": true,
```
Comment (kingthorin) [inline .devcontainer/devcontainer.json:8]: Line 6 needs a comma now? (Blank at 7 could be removed and indent at 8 adjusted?)

Comment (kingthorin) [reply .devcontainer/devcontainer.json:8]: Everything else looks good to me.

Comment (victoriadrake) [reply .devcontainer/devcontainer.json:8]: Oops.
